Oraimunka24015004/src/AlgorithmicRecap.java

```diff
@@ -0,0 +1,188 @@
+import java.util.*;
+
+public class AlgorithmicRecap {
+    public String taskOneReverts(String text) {
+        String result = "";
+        for (int i = text.length(); i >= 1; i--) {
+            char charcter = text.charAt(i - 1);
+            result = result + charcter;
+        }
+        return result;
+    }
+
+    boolean taskTwoIsPalindrome(String text) {
+        return text.equals(taskOneReverts(text));
+    }
+
+    boolean taskThreeIsAnagram(String text, String text2) {
+        Map<Character, Integer> map = new HashMap<>();
+        Map<Character, Integer> map2 = new HashMap<>();
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsKey(character)) {
+                map.put(character, map.get(character) + 1);
+            } else {
+                map.put(character, 1);
+            }
+        }
+        for (int i = 0; i < text2.length(); i++) {
+            char character = text2.charAt(i);
+            if (map2.containsKey(character)) {
+                map2.put(character, map2.get(character) + 1);
+            } else {
+                map2.put(character, 1);
+            }
+        }
+        return map.equals(map2);
+    }
+
+    int taskFourCountNumbers(String text) {
+        int result = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (Character.isDigit(character)) {
+                result += Character.getNumericValue(character);
+            }
+        }
+        return result;
+    }
+
+    int taskFiveGetFactorial(int number) {
+        int result = 1;
+        for (int i = 1; i <= number; i++) {
+            result *= i;
+        }
+        return result;
+    }
+
+    String taskSixGetUniqueChar(String text) {
+        Map<Integer, Character> map = new HashMap<>();
+        int key = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            if (map.containsValue(character)) {
+                map.put(key + 1, character);
+            } else {
+                map.put(key + 1, character);
+            }
+        }
+        return map.toString();
+    }
+
+    StringBuilder taskSevenGetAllUniqueChar(String text) {
+        StringBuilder result = new StringBuilder();
+        List<Character> blacklist = new ArrayList<>();
+        int counter = 0;
+        for (int i = 0; i < text.length(); i++) {
+            char character = text.charAt(i);
+            for (int j = i + 1; j < text.length(); j++) {
+                char tempChar = text.charAt(j);
+                if (tempChar == character) {
+                    counter++;
+                }
+            }
+            if (counter == 0 && !blacklist.contains(character)) {
+                result.append((character));
+            } else {
+                blacklist.add(character);
+                counter = 0;
+            }
+        }
+        return result;
```

Comment:
returning a StringBuilder is kinda odd here

Reply:
the list of characters is better to be returned here instead of the concatenated string